libs/ui/src/screens/swap/components/timer/timer.tsx

```diff
@@ -0,0 +1,49 @@
+import { OnEventFn } from '@rnw-community/shared';
+import React, { FC, useState, useEffect } from 'react';
+import { View } from 'react-native';
+
+import { GetAmountAndRoutesResponse } from '../../../../api/1inch';
+import { Row } from '../../../../components/row/row';
+import { Text } from '../../../../components/text/text';
+
+import { styles } from './timer.styles';
+
+interface Props {
+  getAmountAndRoutes: OnEventFn<void>;
+  routes?: GetAmountAndRoutesResponse['protocols'];
+}
+
+const initialSecondsState = '30';
+
+export const Timer: FC<Props> = ({ getAmountAndRoutes, routes }) => {
+  const [seconds, setSeconds] = useState(initialSecondsState);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setSeconds(currentSeconds => {
+        if (Number(currentSeconds) === 0) {
+          getAmountAndRoutes();
+          clearInterval(interval);
+
+          return '00';
+        }
+
+        return `0${Number(currentSeconds) - 1}`.slice(-2);
```

Comment:
why do you use slice? can you just return ```Number(currentSeconds) - 1```

Reply:
![image](https://user-images.githubusercontent.com/35599200/214312565-bbd20681-0831-4b6a-a666-782b33b6241b.png)
need to add zero for numbers less than 10